common/src/main/java/com/graphicsfuzz/common/util/UpgradeShadingLanguageVersion.java

```diff
@@ -74,37 +86,84 @@
    */
   private List<Stmt> globalVariableInitializers = new ArrayList<>();
 
+  // Variable name renaming map
+  private Map<String, String> variableRename = new HashMap<>();
+
+  // Function name renaming map
+  private Map<String, String> functionRename = new HashMap<>();
+
+  /*
+   * Originally texture fetch functions included information about the kind of sampler
+   * to be used, but later on these were simplified and the remaining texture fetch
+   * functions work with various samplers.
+   */
+  private final String[][] textureFunctionRenames = {
+      {"shadow1D","texture"},
+      {"shadow2D","texture"},
+      {"texture1D","texture"},
+      {"texture2D","texture"},
+      {"texture3D","texture"},
+      {"textureCube","texture"},
+      {"shadow1DProj","textureProj"},
+      {"shadow2DProj","textureProj"},
+      {"texture1DProj","textureProj"},
+      {"texture2DProj","textureProj"},
+      {"texture3DProj","textureProj"},
+      {"shadow1DLod","textureLod"},
+      {"shadow2DLod","textureLod"},
+      {"texture1DLod","textureLod"},
+      {"texture2DLod","textureLod"},
+      {"texture3DLod","textureLod"},
+      {"textureCubeLod","textureLod"},
+      {"shadow1DProjLod","textureProjLod"},
+      {"shadow2DProjLod","textureProjLod"},
+      {"texture1DProjLod","textureProjLod"},
+      {"texture2DProjLod","textureProjLod"},
+      {"texture3DProjLod","textureProjLod"}
+  };
+
+
   private static Namespace parse(String[] args) throws ArgumentParserException {
-    ArgumentParser parser = ArgumentParsers.newArgumentParser("PrettyPrint")
+    ArgumentParser parser = ArgumentParsers.newArgumentParser("UpgradeShadingLanguageVersion")
         .defaultHelp(true)
-        .description("Pretty print a shader.");
+        .description("Upgrade shading language version.");
 
     // Required arguments
     parser.addArgument("shader")
-        .help("Path of shader to be pretty-printed.")
+        .help("Path of shader to be upgraded.")
         .type(File.class);
 
     parser.addArgument("output")
         .help("Target file name.")
         .type(String.class);
 
+    // Optional arguments
+    parser.addArgument("--norename")
+        .help("Do not rename user-defined variables and functions")
+        .action(Arguments.storeTrue());
+
     return parser.parseArgs(args);
   }
 
   public static void main(String[] args) throws IOException, InterruptedException {
 
     try {
       Namespace ns = parse(args);
```

Comment:
I would comment that you are using 'flags' as a bit mask.

Reply:
Done.